### PR TITLE
Handled empty string compares

### PR DESCRIPTION
**Pull request by mh884**
GetMostPreciseType CommonTypes which it will start to check if the left side is "Int64" or right side is "Int64" then it will move to check if the left side is "Double" or right side is "Double" and it goes on, now if you see it start with Int64,Double,Boolean and last thing is Decimal the best way to explain this is by examples before moving there GetMostPreciseType it's a method to check the precise type to convert the right and left side based on the result just keep in mind we start to check from the left to right

'' = 4 -> string
20 = '' -> string
0.5 = '' -> double *
'' = 0.5 -> double*
my fixed will change the double to string as double will performed before string in the foreach loop when you compere take in place between two string

basically, it changes any type to string if any of the sides of compering is not a string

### CODE REVIEW

**Comment by pitermarx** [comment]
I dont get this at all...

**Comment by pitermarx** [comment]
closing due to inactivity
